graylog2-web-interface/src/components/users/EditRole.jsx

```diff
@@ -1,85 +1,77 @@
 import PropTypes from 'prop-types';
 import React from 'react';
-import createReactClass from 'create-react-class';
 import Immutable from 'immutable';
 
 import { Button, Alert, Col, ControlLabel, FormGroup, HelpBlock, Row } from 'components/graylog';
 import { Input } from 'components/bootstrap';
 import PermissionSelector from 'components/users/PermissionSelector';
-import PermissionsMixin from 'util/PermissionsMixin';
 
-const EditRole = createReactClass({
-  displayName: 'EditRole',
-
-  propTypes: {
-    initialRole: PropTypes.object.isRequired,
-    onSave: PropTypes.func.isRequired,
-    cancelEdit: PropTypes.func.isRequired,
-    streams: PropTypes.object.isRequired,
-    dashboards: PropTypes.object.isRequired,
-  },
-
-  mixins: [PermissionsMixin],
-
-  getInitialState() {
+class EditRole extends React.Component {
+  constructor(props) {
+    super(props);
     const { initialRole } = this.props;
-
     let role = initialRole;
-    if (role === null) {
-      // for the create dialog
+    if (!role) {
       role = {
         name: null,
         description: null,
         permissions: [],
       };
     }
-    return {
+    this.state = {
       role,
       initialName: this._safeRoleName(initialRole),
     };
-  },
+  }
 
-  componentWillReceiveProps(newProps) {
-    this.setState({ role: newProps.initialRole, initialName: this._safeRoleName(newProps.initialRole) });
-  },
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const {
+      role = {
```

Comment:
Since we use `getDerivedStateFromProps` to initialize an "empty" role, and `getDerivedStateFromProps` is called before the first render and before any subsequent update is rendered, I think the constructor can be simplified and doesn't need to handle the default case.

So I think we can simplify this logic doing the following:
1. In the constructor, we set the role to the `initialRole`, since we are also using that to get the `initialName`
2. In `getDerivedStateFromProps` we check if `prevState.role` is set, and if it is not set we initialize it to the "empty" role

Reply:
Ok for that, added the update